subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/worker/device/adb/RetryAction.kt

```diff
@@ -0,0 +1,34 @@
+package com.avito.runner.service.worker.device.adb
+
+import com.avito.time.TimeProvider
+import java.util.concurrent.TimeUnit
+
+class RetryAction(private val timeProvider: TimeProvider) {
+
+    fun <T> retry(
+        retriesCount: Int,
+        delaySeconds: Long = 1,
+        onError: (attempt: Int, throwable: Throwable, durationMs: Long) -> Unit = { _, _, _ -> },
```

Comment:
onError onFailure will make no sense after a couple of months 
Let's rename them to onAttemptError onLastAttemptError

Reply:
https://github.com/avito-tech/avito-android/pull/760/files
I swear it was your idea

Reply:
:( In the concrete function, I realized that I feel onFailure and onError the same but maybe I'm wrong

Reply:
i'll left it as it is for now